src/main/java/org/kiwiproject/beta/base/KiwiStrings2.java

```diff
@@ -83,4 +106,193 @@ public static String replaceNullCharacters(@Nullable String str, String replacem
                 .map(s -> NULL_CHAR_PATTERN.matcher(s).replaceAll(replacement))
                 .orElse(defaultValue);
     }
+
+    /**
+     * Generate a set of three strings (or two when the input is only one character) whose characters match the input
+     * string but are random upper and lower case using the default {@link Locale}.
+     *
+     * @param input the input string to generate random case variants for
+     * @return an unmodifiable set containing three random case variants of the input string
+     */
+    public static Set<String> randomCaseVariants(@NonNull String input) {
+        return randomCaseVariants(input, Locale.getDefault());
+    }
+
+    /**
+     * Generate a set of three strings (or two when the input is only one character) whose characters match the input
+     * string but are random upper and lower case using the given {@link Locale}.
+     *
+     * @param input  the input string to generate random case variants for
+     * @param locale the Locale to use to conversion to upper and lower case
+     * @return an unmodifiable set containing three random case variants of the input string
+     */
+    public static Set<String> randomCaseVariants(@NonNull String input, @NonNull Locale locale) {
+        return randomCaseVariants(input, 3, locale);
+    }
+
+    /**
+     * Get the current maximum number of random strings that can be generated using
+     * {@link #randomCaseVariants(String, int)} and {@link #randomCaseVariants(String, int, Locale)}.
+     *
+     * @return the maximum number of random strings that can be generated
+     */
+    @Synchronized
```

Comment:
why does reading the value have to be synchronized?

Reply:
What if some other thread is writing a new value while your thread is reading it?

Reply:
you might get an outdated value